### PR TITLE
secscan: Don't save result if returned state is unknown (PROJQUAY-2939)

### PR DESCRIPTION
**Pull request by crozzy**
Currently it is possible that Clair returns a state that Quay doesn't
know about and the previous logic will save this as a completed
manifest.

Signed-off-by: crozzy <joseph.crosland@gmail.com>

### CODE REVIEW

**Comment by crozzy** [comment]
Need to add a test case for this

File: data/secscan_model/secscan_v4_model.py [hunk 1]
```diff
@@ -294,6 +294,14 @@ def mark_manifest_unsupported(manifest):
                 logger.exception("Failed to perform indexing, security scanner API error")
                 continue
 
+            if report["state"] == IndexReportState.Index_Finished:
+                index_status = IndexStatus.COMPLETED
+            elif report["state"] == IndexReportState.Index_Error:
+                index_status = IndexStatus.FAILED
+            else:
+                # Unknown state don't save anything
+                continue
```
Comment (kleesc) [inline data/secscan_model/secscan_v4_model.py:303]: What are these states? Is the client expected to retry in these cases?

Comment (crozzy) [reply data/secscan_model/secscan_v4_model.py:303]: If something goes wrong during the indexing the last state is returned, i.e. "scanLayers", these should be retried and when they are, Clair will pick up where it left off.

**Comment by crozzy** [comment]
CI needs this merged to pass https://github.com/quay/quay/pull/1057/files

**Review by kleesc — APPROVED** [review]
LGTM

**Comment by hdonnay** [comment]
rebased and fixed the formatting complaint

**Review by kleesc — APPROVED** [review]
LGTM

**Pull request merged**